### DISCUSSION TITLE
How can you create dynamic headings with code using Python?

### DISCUSSION THREAD

**Discussion by SamEdwardes** [quarto-dev/quarto-cli#11201] [Q&A] [answered]
## Background

I am trying to dynamically create markdown based on data that I am processing in Python. It is mostly working, except the headings that I generate do not show up in the table of contents.

Is it possible to get them to appear in the TOC?

A few things to consider:

- The anchor tag is still generated. I can link to the heading. It just does not appear in the TOC.
- Note that it does work when I use inline code. This will appear in the TOC.

## Reprex

Take the following document:

``````qmd
---
title: "Reprex"
format: 
    html:
        toc: true
        toc-depth: 6
---

```{python}
from IPython.display import Markdown
```

## heading 2.1

## heading 3.1

```{python}
headings = [
    "### Heading 3.2",
    "### Heading 3.3",
    "## Heading 4.1",
    "### Heading 4.2",
]

for i in headings:
    display(Markdown(i))
```

`{python} Markdown("## Heading 5.1")`

`{python} Markdown("### Heading 5.2")`

`{python} Markdown("### Heading 5.3")`
``````

![Screenshot 2024-10-25 at 16 01 37@2x](https://github.com/user-attachments/assets/42ebe819-49c2-4225-8bf0-1a32079a848b)

## System info

- Quarto: 1.5.47
- Mac OS: 15.0.1 (24A348)
- MacBook Pro M1

**Comment by mcanouil** [comment, marked as answer]
Use `output: asis` when using `display()`.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Reprex"
format: 
    html:
        toc: true
        toc-depth: 6
---

```{python}
from IPython.display import Markdown
```

## heading 2.1

## heading 3.1

```{python}
#| output: asis
headings = [
    "### Heading 3.2",
    "### Heading 3.3",
    "## Heading 4.1",
    "### Heading 4.2",
]

for i in headings:
    display(Markdown(i))
```

`{python} Markdown("## Heading 5.1")`

`{python} Markdown("### Heading 5.2")`

`{python} Markdown("### Heading 5.3")`

````

</td><td>

<img width="653" alt="image" src="https://github.com/user-attachments/assets/d60904f8-ad16-4110-8234-0f7db96e5a42">

</td></tr>
</table>

**Reply by SamEdwardes** [reply]
Thank you I will give this a try!

**Reply by SamEdwardes** [reply]
Confirmed this worked. Thank you for the advice :)
